src/components/tooltip/Tooltip.tsx

```diff
@@ -37,10 +37,29 @@ const Tooltip = (props: TooltipProps) => {
       const rect = target.getBoundingClientRect();
       const tooltipRect = tooltipRef.current.getBoundingClientRect();
 
-      setPosition({
-        top: rect.top + rect.height / 2 - tooltipRect.height / 2,
-        left: rect.right + 10,
-      });
+      if (rect.right + tooltipRect.width > window.innerWidth) {
+        // If the tooltip box is past the right edge of the page, position it to the top
+        setIsOverflowing(true);
+        setPosition({
+          top: rect.top - rect.height / 2 - tooltipRect.height,
+          left: window.innerWidth / 2 - tooltipRect.width / 2,
+        });
+        // Calculate margin so triangle will be pointing to info icon
+        const calcTriangleMargin = `${
+          window.innerWidth / 2 - rect.right + 5
+        }px`;
+        document.documentElement.style.setProperty(
+          "--tooltip-margin-right",
+          calcTriangleMargin,
+        );
```

Comment:
Can we apply this in the computed `style` below, rather than mutating the document?

Reply:
I've removed the document mutation, but I'm not certain if my new approach is exactly what you suggested.

Reply:
Ty! Almost exactly what I was looking for -- I pushed a very minor tweak to also do the non-above margin with it as well for consistency